base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/ConfigHandler.java

```diff
@@ -91,4 +96,51 @@ protected IWorkbenchWindow activeWindow() {
      */
 	public abstract void canWriteChanged(boolean canWrite);
 	
+    /**
+     * Implements the required eclipse interface.
+     * 
+     * Declared as final here, implement executeAndCatchErrors instead so that
+     * we can "hook into" any errors that occur in any of these dialogs and do
+     * something sensible with them.
+     * 
+     * @param event
+     *            the execution event
+     * @return null (ignored by eclipse framework)
+     */
+    @Override
+    public final Object execute(ExecutionEvent event) {
+        try {
+            safeExecute(event);
+        } catch (RuntimeException e) {
+            onError(e);
+        }
+        return null;
+    }
+
+    /**
+     * Executes an eclipse event. Any runtime errors thrown by this method will
+     * be caught by the execute() method and cause the onError() handler to be
+     * called.
+     * 
+     * @param event
+     *            the eclipse event
+     */
+    public abstract void safeExecute(ExecutionEvent event);
+
+    /**
+     * Error handler. Override this to provide specific error handling
+     * behaviour.
+     * 
+     * @param ex
+     *            the exception that was thrown from the handler
+     */
+    public void onError(Exception ex) {
+        Logger.getGlobal().log(Level.SEVERE, "Exception while executing configserver command", ex);
+
+        MessageDialog.openError(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(),
+                "Error",
+                "An error occured while performing that action. Error was:\n" + ex.toString()
+                        + "\nPlease see the error log for more details.");
```

Comment:
Make this something like. "An error occurred in the GUI please report the following problem to the computing group with a description of what you were trying to do. Thank you."